tests/test_setup.py

```diff
@@ -66,6 +67,29 @@ def test_setup_already_tracked_volume(self):
         self.assertTrue(os.path.exists(self.snap_device))
         self.assertIsNotNone(elastio_snap.info(self.minor))
 
+    def test_setup_check_cow_size_zero_fallocate(self):
+        self.assertEqual(elastio_snap.setup(self.minor, self.device, self.cow_full_path, fallocated_space=0), 0)
+        self.addCleanup(elastio_snap.destroy, self.minor)
+
+        page_size = util.os_page_size()
+        cow_file_size_factor = 0.1 # 10% by default if `fallocated_space` is 0
+        cow_file_size = util.dev_size_bytes(self.device) * cow_file_size_factor
+
+        # rounding up aligned to the PAGE_SIZE
+        cow_file_size = int(math.ceil(cow_file_size / page_size) * page_size);
+        self.assertEqual(os.stat(self.cow_full_path).st_size, cow_file_size)
```

Comment:
It would be great to add one move comparison of `cow_file_size` with `fallocate` value from `/proc/elastio-snap-info`.

Reply:
Done.